lib/ransack/adapters/active_record/base.rb

```diff
@@ -45,7 +45,7 @@ def ransackable_attributes(auth_object = nil)
         # For overriding with a whitelist array of strings.
         #
         def ransackable_associations(auth_object = nil)
-          reflect_on_all_associations.map { |a| a.name.to_s }
+          @ransackable_associations ||= reflect_on_all_associations.map { |a| a.name.to_s }
```

Comment:
> Technically, new values could be added by code at runtime -- they would need to invalidate the cache.

Depending on your definition of "runtime", `PaperTrail` does this (adds associations to your models).